### PR TITLE
[#34] CLI command to generate commit message from branch name

### PR DESCRIPTION
**Pull request by weiland**
This PR was already created with this modification.

### CODE REVIEW

**Comment by bitboxer** [comment]
Can you add an explanation for this in the readme?

File: src/common/popup/utils/format.js [hunk 1]
```diff
@@ -14,8 +14,13 @@ function branch(ticket) {
   return `${ticket.type || 'feature'}/${ticket.id}-${normalize(ticket.title)}`;
 }
 
+function command(ticket) {
+  return `git checkout -b ${branch(ticket)}  && git commit --allow-empty -m"${commit(ticket)}"`;
```
Comment (pmeinhardt) [inline src/common/popup/utils/format.js]: There should be a space between `-m` and the message. It'll work like this, but for extra style points… 😝 (https://git-scm.com/docs/git-commit) (Sorry for nitpicking).

Comment (tessi) [inline src/common/popup/utils/format.js]: i gave a bad example in my proposal-issue ;) (for some reason i never write that space, to save a byte)

**Comment by pmeinhardt** [comment]
Unfortunately I could not find a JavaScript package, that implements
shell-escaping. The `shellquote` function implemented here currently
does the following:

  1. Wraps argument in single-quotes (or returns `''` if none given)
  2. Escapes any single-quotes in the input string

Shell-quoting is applied to both the branch name and the commit message
generated by Tickety-Tick.

I do not at all like implementing such a crucial function here and will
do some more research into available candidate implementations & further
potential issues (e.g. Unicode).

@bitboxer, @tessi: Maybe you have some further knowledge on this matter?

I've looked at what Ruby's [Shellwords#escape] does (though not directly applicable) and a couple of other related pieces of code (e.g. [mathiasbynens/jsesc](https://github.com/mathiasbynens/jsesc)).

Bash manual on quoting: https://www.gnu.org/software/bash/manual/html_node/Single-Quotes.html

**Comment by pmeinhardt** [comment]
Here's an issue that you can play around with (if you're building the extension from source).

**Comment by bitboxer** [comment]
I think it's okay enough for now. Don't see any other JS library, too.

**Pull request merged**